helium-lib/src/client.rs

```diff
@@ -23,17 +41,195 @@ pub static SOLANA_URL_DEVNET: &str = "https://solana-rpc.web.test-helium.com?ses
 pub static SOLANA_URL_MAINNET_ENV: &str = "SOLANA_MAINNET_URL";
 pub static SOLANA_URL_DEVNET_ENV: &str = "SOLANA_DEVNET_URL";
 
-pub use solana_client::nonblocking::rpc_client::RpcClient as SolanaRpcClient;
+static USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"),);
+
+#[derive(Clone)]
+pub struct SolanaClient {
+    pub inner: Arc<SolanaRpcClient>,
+    pub base_url: String,
+    pub wallet: Option<Arc<Keypair>>,
+}
+
+impl Default for SolanaClient {
+    fn default() -> Self {
+        Self::new(SOLANA_URL_MAINNET, None).unwrap()
+    }
+}
+
+impl SolanaClient {
+    pub fn new(url: &str, wallet_path: Option<PathBuf>) -> Result<Self, Error> {
+        let client = Arc::new(
+            solana_client::nonblocking::rpc_client::RpcClient::new_with_commitment(
+                url.to_string(),
+                CommitmentConfig::finalized(),
+            ),
+        );
+
+        let wallet = if let Some(path) = wallet_path {
+            Some(Arc::new(
+                Keypair::read_from_file(&path).map_err(Error::from)?,
+            ))
+        } else {
+            None
+        };
+
+        Ok(Self {
+            inner: client,
+            base_url: url.to_string(),
+            wallet,
+        })
+    }
+
+    pub fn solana_rpc_client(&self) -> Arc<SolanaRpcClient> {
+        self.inner.clone()
+    }
+
+    pub fn ws_url(&self) -> String {
+        self.base_url
+            .replace("https", "wss")
+            .replace("http", "ws")
+            .replace("127.0.0.1:8899", "127.0.0.1:8900")
+    }
+
+    pub fn wallet(&self) -> Result<Pubkey, Error> {
+        self.wallet
+            .as_ref()
+            .map(|wallet| wallet.pubkey())
+            .ok_or_else(|| Error::Other("Wallet not configured".to_string()))
+    }
+
+    pub async fn send_instructions(
```

Comment:
yeah it could be standalone, I think we would only need to pass client to it and client could pull the wallet off itself. Will ad that to the `transaction.rs` file